Sources/Player/PlaybackEngine/Internal/AVQueuePlayer/AssetCache.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+// MARK: - AssetCacheState
+
+public struct AssetCacheState {
+	public enum AssetCacheStatus: Equatable {
+		case notCached
+		case cached(cachedURL: URL)
+	}
+
+	public let key: String
+	public let status: AssetCacheStatus
+
+	public init(key: String, status: AssetCacheState.AssetCacheStatus) {
+		self.key = key
+		self.status = status
+	}
+}
+
+// MARK: - AssetCache
+
+public final class AssetCache {
+	private var userDefaults: UserDefaultsClient
+
+	public init(userDefaults: UserDefaultsClient = UserDefaultsClient.live()) {
+		self.userDefaults = userDefaults
+	}
+
+	public func get(_ key: String) -> AssetCacheState {
+		var bookmarkIsStale = false
```

Comment:
[suggestion] I think this alternative reads a bit more as an assertion:
```suggestion
		var isBookmarkStale = false
```